.github/workflows/audit.yml

```
@@ -0,0 +1,19 @@
+name: Security Audit
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

```suggestion
    branches: [ master ]
    paths: 
      - '**/Cargo.toml'
      - '**/Cargo.lock'
```

No need to trigger the workflow if the commit doesn't affect the dependencies.